include/fsutils/ipcfg.h

```diff
@@ -31,23 +31,35 @@
  * Public Types
  ****************************************************************************/
 
-/* Values for the BOOTPROTO setting */
+/* Values for the record type field */
 
-enum ipcfg_bootproto_e
+enum ipcfg_iptype_e
 {
-  BOOTPROTO_NONE     = 0, /* No protocol assigned */
-  BOOTPROTO_STATIC   = 1, /* Use static IP */
-  BOOTPROTO_DHCP     = 2, /* Use DHCP */
-  BOOTPROTO_FALLBACK = 3  /* Use DHCP with fall back static IP */
+  IPTYPE_IPv4        = 0, /* IPv4 configuration */
+  IPTYPE_IPv6        = 1  /* IPv6 configuration */
+};
+
+/* Generic IP Configuration record header.  Must be cast compatible with
+ * both struct ipv4cfg_s and struct ipv6cfg_s
+ */
+
+struct ipcfg_s
+{
+  uint8_t next;                 /* Offset to the next IP configuration record */
+  enum ipcfg_iptype_e type;     /* Must be IPTYPE_IPv4 */
```

Comment:
@patacongo did you consider using the AF_INET/AF_INET6 here or did you want the extra abstraction>?

Reply:
Done.  sa_family_t is a little larger, but more straightforward.